src/roles/integration.test.ts

```diff
@@ -1,12 +1,288 @@
+    },
+  },
+  {
+    roleName: 'nodes-minimal',
+    permissions: weaviate.permissions.nodes.minimal({
+      read: true,
+    }),
+    expected: {
+      name: 'nodes-minimal',
+      ...emptyPermissions,
+      nodesPermissions: [{ collection: '*', verbosity: 'minimal', actions: ['read_nodes'] }],
+    },
+  },
+  {
+    roleName: 'roles',
+    permissions: weaviate.permissions.roles({
+      role: 'some-role',
+      ...crud,
+    }),
+    expected: {
+      name: 'roles',
+      ...emptyPermissions,
+      rolesPermissions: [{ role: 'some-role', actions: rolesActions }],
+    },
+  },
+  {
+    roleName: 'tenants-st',
+    permissions: weaviate.permissions.tenants({
+      collection: 'some-collection',
+      ...crud,
+    }),
+    expected: {
+      name: 'tenants-st',
+      ...emptyPermissions,
+      tenantsPermissions: [
+        {
+          collection: 'Some-collection',
+          tenant: '*',
+          actions: tenantsActions,
+        },
+      ],
+    },
+  },
+  {
+    roleName: 'tenants-mt',
+    permissions: weaviate.permissions.tenants({
+      collection: 'some-collection',
+      tenant: 'some-tenant',
+      ...crud,
+    }),
+    expected: {
+      name: 'tenants-mt',
+      ...emptyPermissions,
+      tenantsPermissions: [
+        {
+          collection: 'Some-collection',
+          tenant: 'some-tenant',
+          actions: tenantsActions,
+        },
+      ],
+    },
+  },
+  {
+    roleName: 'tenants-mt-mixed',
+    permissions: weaviate.permissions.tenants({
+      collection: ['some-collection', 'another-collection'],
+      tenant: ['some-tenant', 'another-tenant'],
+      ...crud,
+    }),
+    expected: {
+      name: 'tenants-mt-mixed',
+      ...emptyPermissions,
+      tenantsPermissions: [
+        {
+          collection: 'Some-collection',
+          tenant: 'some-tenant',
+          actions: tenantsActions,
+        },
+        {
+          collection: 'Some-collection',
+          tenant: 'another-tenant',
+          actions: tenantsActions,
+        },
+        {
+          collection: 'Another-collection',
+          tenant: 'some-tenant',
+          actions: tenantsActions,
+        },
+        {
+          collection: 'Another-collection',
+          tenant: 'another-tenant',
+          actions: tenantsActions,
+        },
+      ],
+    },
+  },
+  {
+    roleName: 'users',
+    permissions: weaviate.permissions.users({
+      user: 'some-user',
+      assignAndRevoke: true,
+      read: true,
+    }),
+    expected: {
+      name: 'users',
+      ...emptyPermissions,
+      usersPermissions: [{ users: 'some-user', actions: ['assign_and_revoke_users', 'read_users'] }],
+    },
+  },
+];
+
+const maybe = DbVersion.fromString(`v${process.env.WEAVIATE_VERSION!}`).isAtLeast(1, 29, 0)
   ? describe
   : describe.skip;
 
-only('Integration testing of the roles namespace', () => {
+maybe('Integration testing of the roles namespace', () => {
```

Comment:
You could use [`describe.skipIf`](https://vitest.dev/api/#describe-skipif) if it looks more readable:

```ts
const version = DbVersion.fromString(...);
describe.skipIf(!version.isAtLeast(1, 29, 0), 'Integration testing ...' () => {
``` 

There's a good alias for it too: `suite.skipIf`

Reply:
This doesn't appear available in the Jest version of the client. I found [this](https://github.com/jestjs/jest/issues/3652#issuecomment-303964364) but it looks like it was never merged. Do you have a source for `describe.skipIf`?

Reply:
Right, I should've checked the version we're using rn 😅